safenode/src/protocol/types/wallet/local_store.rs

```diff
@@ -0,0 +1,344 @@
+
+        self.deposit(change_dbc.into_iter().collect());
+        self.wallet.spent_dbcs.append(&mut spent_dbcs);
+        self.wallet.output_history.extend(created_dbcs.clone());
+
+        Ok(created_dbcs)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::{DepositWallet, LocalDepositWallet, Result, SendClient};
+
+    use crate::protocol::types::{
+        dbc_genesis::{create_genesis_dbc, GenesisResult, GENESIS_DBC_AMOUNT},
+        transfers::{create_transfer, Outputs as TransferDetails},
+        wallet::{LocalSendWallet, SendWallet},
+    };
+
+    use sn_dbc::{Dbc, DbcIdSource, DerivedKey, MainKey, PublicAddress, Token};
+
+    /// -----------------------------------
+    /// <-------> DepositWallet <--------->
+    /// -----------------------------------
+
+    #[test]
+    fn deposit_wallet_basics() -> Result<()> {
+        let main_key = MainKey::random();
+        let public_address = main_key.public_address();
+        let local_wallet: LocalDepositWallet = DepositWallet::new(main_key);
+
+        assert_eq!(public_address, local_wallet.address());
+        assert_eq!(
+            public_address,
+            local_wallet.new_dbc_address().public_address
+        );
+        assert_eq!(Token::zero(), local_wallet.balance());
+
+        assert!(local_wallet.available_dbcs.is_empty());
+        assert!(local_wallet.output_history.is_empty());
+        assert!(local_wallet.spent_dbcs.is_empty());
+
+        Ok(())
+    }
+
+    #[test]
+    fn deposit_empty_list_does_nothing() -> Result<()> {
+        let mut local_wallet: LocalDepositWallet = DepositWallet::new(MainKey::random());
+
+        local_wallet.deposit(vec![]);
+
+        assert_eq!(Token::zero(), local_wallet.balance());
+
+        assert!(local_wallet.available_dbcs.is_empty());
+        assert!(local_wallet.output_history.is_empty());
+        assert!(local_wallet.spent_dbcs.is_empty());
+
+        Ok(())
+    }
+
+    #[test]
+    #[allow(clippy::result_large_err)]
+    fn deposit_adds_dbcs_that_belongs_to_the_wallet() -> GenesisResult<()> {
+        let genesis_key = MainKey::random();
+        let genesis = create_genesis_dbc(&genesis_key)?;
+
+        let mut local_wallet: LocalDepositWallet = DepositWallet::new(genesis_key);
+
+        local_wallet.deposit(vec![genesis]);
+
+        assert_eq!(GENESIS_DBC_AMOUNT, local_wallet.balance().as_nano());
+
+        Ok(())
+    }
+
+    #[test]
+    #[allow(clippy::result_large_err)]
+    fn deposit_does_not_add_dbcs_not_belonging_to_the_wallet() -> GenesisResult<()> {
+        let genesis_key = MainKey::random();
+        let genesis = create_genesis_dbc(&genesis_key)?;
+
+        let wallet_key = MainKey::random();
+        let mut local_wallet: LocalDepositWallet = DepositWallet::new(wallet_key);
+
+        local_wallet.deposit(vec![genesis]);
+
+        assert_eq!(Token::zero(), local_wallet.balance());
+
+        Ok(())
+    }
+
+    /// --------------------------------
+    /// <-------> SendWallet <--------->
```

Comment:
what does it mean for `SendWallet` ?
transfer the ownship of the wallet to other,
or send the whole balance of a wallet to other,
or wallet send some balance of it to other ?

Reply:
It's described in the comments. A send wallet can send. A deposit wallet can only deposit.